### PR TITLE
Feature/sg 000 yolo nas dropout support

### PR DESCRIPTION
**Pull request by BloodAxe**
Adds the ability to specify dropout parameters in the YoloNAS model.
Compatibility with pretrained checkpoints remains intact. 
Default values of dropout probas are set to zero to keep existing training recipies intact.
But if users wants, he can customize these in arch params config.

### CODE REVIEW

File: src/super_gradients/modules/dropout_layers.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    >>> class ResNetBlock(nn.Module):
+    >>>   def __init__(self, ..., drop_path_rate:float):
+    >>>     self.drop_path = DropPath(drop_path_rate)
+    >>>
+    >>>   def forward(self, x):
+    >>>     return self.drop_path(x) + self.conv_bn_act(x)
+    """
+
+    def __init__(self, drop_prob: float = 0.0, scale_by_keep: bool = True):
+        super(DropPath, self).__init__()
```
Comment (Louis-Dupont) [inline src/super_gradients/modules/dropout_layers.py:35]: If you can just add description of params

Comment (BloodAxe) [reply src/super_gradients/modules/dropout_layers.py]: Thanks for the suggestion. While adding docs I realized we in fact already have the DropPath implementation. So I update it a little bit with docs and removed mine. But in terms of numerical output they are the same

**Review by Louis-Dupont — APPROVED** [review]
LGTM
